metaflow/cli.py

```diff
@@ -649,16 +649,39 @@ def run(obj,
               default=DEFAULT_RUN_NAME,
               help="name assigned to the new KFP run"
               )
+@click.option('--namespace',
+              'namespace',
+              default=KFP_SDK_NAMESPACE,
+              help="namespace of your run in KFP."
+              )
+@click.option('--api-namespace',
+              'api_namespace',
+              default=KFP_SDK_API_NAMESPACE,
+              help="namespace where the API service is run."
+              )
+@click.option('--userid',
+              'userid',
+              default=KFP_SDK_USERID,
+              help="your user ID (your email)."
+              )
 @click.pass_obj
 def run_on_kfp(obj,
         code_url=DEFAULT_FLOW_CODE_URL,
         experiment_name=DEFAULT_EXPERIMENT_NAME,
-        run_name=DEFAULT_RUN_NAME
+        run_name=DEFAULT_RUN_NAME,
+        namespace=KFP_SDK_NAMESPACE,
+        api_namespace=KFP_SDK_API_NAMESPACE,
+        userid=KFP_SDK_USERID
         ):
 
-    run_pipeline_result = create_run_on_kfp(obj.graph, code_url, experiment_name, run_name)
+    if namespace is None or userid is None:
+        raise Exception("Both namespace and userid must be defined, either through the CLI or as environment variables.")
```

Comment:
style: please put a new line after an IF statement block, this would make it easier to read, and less likely to mix blocks of code into the if statement.

Reply:
Done